src/piadm.sh

```diff
@@ -486,20 +488,50 @@ update_boot_sectors() {
 
 	some=0
 	for a in "${boot_devices[@]}"; do
-		# Plow through devices, even if some fail.  installboot also
-		# does loader-into-EFI-System-Partition this way.  Trailing /
-		# is important in the -b argument because boot is actually a
-		# symlink.
-		installboot -m -b /${bootfs}/boot/ /${bootfs}/boot/pmbr \
-			/${bootfs}/boot/gptzfsboot \
-			/dev/rdsk/${a}${suffix} > /dev/null 2>&1 || \
-			eecho "WARNING: Can't installboot on ${a}${suffix}"
+		if [[ "$flag" == "-d" ]]; then
+			if [[ "$suffix" == "s0" ]]; then
+				# BIOS boot, we don't care.
+				continue
+			fi
+			# otherwise mount the ESP and trash it.
+			if [[ "$tdir" == "" ]]; then
+				tdir=`mktemp -d`
+			fi
+			mount -F pcfs /dev/dsk/${a}s0 ${tdir}
+			if [[ $? -ne 0 ]]; then
+				eecho "disk $a has no PCFS ESP, it seems"
+				continue
+			fi
+			# Just take out the EFI directory, in case someone
+			# is using it for something ELSE also.
+			/bin/rm -rf ${tdir}/EFI
+			umount ${tdir}
+			# If we make it here, at least some disks had
+			# ESP and we managed to clean them out.  "some" below
+			# will get set.
+		else
+			# Plow through devices, even if some fail.
+			# installboot also does
+			# loader-into-EFI-System-Partition this way.
+			# Trailing / is important in the -b argument
+			# because boot is actually a symlink.
+			installboot -m -b /${bootfs}/boot/ \
+				/${bootfs}/boot/pmbr \
+				/${bootfs}/boot/gptzfsboot \
+				/dev/rdsk/${a}${suffix} > /dev/null 2>&1 || \
+				eecho \
+				"WARNING: Can't installboot on ${a}${suffix}"
+		fi
+
 		if [[ $? -eq 0 ]]; then
 			some=1
 		fi
 	done
+
+	# Partial success (altering some of the pool's disks) is good
+	# enough for command success.
 	if [[ $some -eq 0 ]]; then
-		fatal "Could not installboot(1M) on ANY vdevs of pool $2"
+		fatal "Could not make alterations on ANY vdevs of pool $2"
```

Comment:
I would just say "update" or "modify" here.

Reply:
Okay, "modify" it is.